# ISCB-Africa ASBCB 2025

**Venue:** Lagoon Beach Hotel & Conference Center, Cape Town, South Africa

**Website:** https://www.iscb.org/africa2025/home

**Date of the session:** April 10, 2025 12:00-16:00 SAST

**Instructors/Affiliation:** 
Loni Taylor, Meharry Medical College, Nashville, TN, USA.
Bishnu Sarker, Meharry Medical College, Nashville, TN, USA.
Animesh Acharjee, University of Birmingham, UK.

Please cite this tutorial as:

**Bishnu Sarker, Loni Taylor, Animesh Acharjee (2025, April). Tutorial VT1: Multiomics Data Integration using Graph Based Machine Learning. In ISCB-Africa ASBCB 2025 at Lagoon Beach Hotel & Conference Center, April 10, 2025, Cape Town, South Africa.**

# Tutorial VT1: Multiomics Data Integration using Graph Based Machine Learning
### Overview:
In recent years, the field of systems biology has made significant strides, driven by the explosion of multiomics data. Genomics, proteomics, transcriptomics, and metabolomics provide complementary insights into the molecular underpinnings of biology, but analyzing these data sets in isolation often limits our ability to uncover complex biological relationships. The true potential of multiomics lies in integrating these diverse data layers to create a holistic understanding of cellular and organismal processes.

However, integrating and analyzing multiomics data is no small task. The sheer volume, complexity, and heterogeneity of these datasets present unique challenges that require sophisticated computational approaches. In this session, we will delve into some of the most powerful techniques for tackling these challenges, including Non-negative Matrix Factorization (NMF), machine learning, and Graph Neural Networks (GNNs). These methods allow us to uncover hidden patterns and relationships across biological layers, which can lead to insights into disease mechanisms, drug responses, and the identification of novel biomarkers.

By applying these advanced tools to real-world multiomics datasets, this tutorial will not only enhance your computational skills but also provide a deeper understanding of how these techniques can be used to drive biological discoveries and improve clinical outcomes.

This repository provides a tutorial to introduce participants to the integration of multiomics data from genomics, proteomics, transcriptomics, and metabolomics, focusing on computational approaches to uncover hidden relationships between biological entities. It will cover techniques such as Non-negative Matrix Factorization (NMF), machine learning, and Graph Neural Networks (GNNs) to model multi-layered biological interactions and predict biological outcomes such as disease classification, drug responses, and biomarker discovery. Attendees will gain hands-on experience in processing and analyzing real-world multiomics datasets using open-source tools such as Python, pandas, scikit-learn.

### Learning Objectives
By the end of this tutorial, attendees will be able to:

1.  Understand key computational approaches for integrating and analyzing multiomics data, including NMF, machine learning, and GNNs.
2.  Apply open-source tools to implement predictive models for disease classification, biomarker identification, and drug response analysis.
3.  Evaluate model performance and interpret results to derive meaningful biological insights.

### Intended Audience and Level
This tutorial is designed for intermediate to advanced learners with an interest in multiomics data integration and its applications in machine learning. It is ideal for bioinformaticians, computational biologists, data scientists, and machine learning practitioners who want to expand their knowledge of graph-based methods in multiomics analysis.

The tutorial is targeted at professionals, researchers, and graduate students with at least a basic understanding of:
+ Omics Data (Genomics, Proteomics, Metabolomics)
+ Machine Learning, especially Neural Networks
+ Programming (Python, familiarity with ML libraries like PyTorch or TensorFlow)

**There are three code notebooks that will be used in this tutorial. Using these code notebooks we will cover:**
1.	Non-negative Matrix Factorization (NMF)
2.	Using NMF for Biomarker Discovery
3.	Graph Neural Networks (GNNs)

### Instructions
The participants are requested to follow the following steps to prepare their work environment.

##### **Minimum Requirements:**
+ A computer
+ High-speed internet.
+ A working python environment
+ Minimum working knowledge of Python, particularly basic looping, lists, array, and tensors. A refresher on Python will be provided as a part of the tutorial. However, we recommend a quick refresher on PyTorch and Scikit-Learn Python packages for ML model development.

##### **Setting up the environment:**
+ Follow the instruction during the session.

# Outline of the Workshop
### Session 1: 12:00-12:30 hrs SAST | Introduction to Multiomics Data Integration.
+ An introduction to the concept of multiomics, highlighting the different types of omics data and the importance of integrating them to uncover deeper biological insights.

### Session 2: 12:30-13:00 hrs SAST | Key Computational Approaches for Multiomics Integration.
+ An overview of key computational methods, including NMF, machine learning, and GNNs, used to integrate and analyze multiomics data.

### 13:00-13:15 hrs SAST - 15 minutes Break/Q&A/Setting up workspace

### Session 3: 13:15-14:15 hrs SAST | Non-negative Matrix Factorization (NMF) for Multiomics Data.
+ A look at how Non-negative Matrix Factorization (NMF) can be applied to decompose complex multiomics data into meaningful biological components.
+ Exploration of how NMF can be leveraged to identify potential biomarkers for disease diagnosis, prognosis, or therapeutic targeting.

### 14:15-14:30 hrs SAST - 15 minutes Break/Q&A

### Session 4: 14:30-15:30 hrs SAST | Similarity Network Based Multi-omics Integration
+ An hands on introduction to graph based integration and their ability to model complex relationships and interactions in multiomics data.
+ A practical guide on how Similarity Network  can be applied to multiomics data to identify biomarkers.

### 15:30-16:00 hrs SAST- 15 Concluding Remarks and Q&A

# Speakers
## 1. Loni Taylor<br>
Loni Taylor is a third year PhD candidate in Biomedical Data Science student in the School of Applied Computational Sciences at Meharry Medical College. She is an active practitioner of data analytics and technology for science enthusiast. Her research focus is in task automation and adaptive learning. She has expertise in machine learning and technology integration.

## 2. Bishnu Sarker<br>
Bishnu Sarker is an Assistant Professor of Computer Science and Data Science at Meharry Medical College, Nashville, TN, USA. He is the Principal Investigator of Sarker’s lab Omics Mining and Algorithmic Reasoning (L’OMAR). His research focus is on applying AI, deep learning, natural language processing (NLP), and graph-based reasoning approaches to effectively describe proteins numerically and to infer their functional characteristics from complex, heterogeneous, and interconnected biomedical data. He received his BS from Khulna University of Engineering and Technology, Bangladesh; MS from Sorbonne University, France; and PhD from INRIA, France. During his PhD he spent a winter at MILA - Quebec AI Institute and University of Montreal, Canada, as a visiting researcher with the DrEAM mobility grant from University of Lorraine.

## 3. Animesh Acharjee<br>
Dr. Acharjee did his undergraduate degree in Electrical Engineering from North Eastern Regional Institute of Technology (NERIST), Itanagar, India and Masters in Bioinformatics from Institute of Bioinformatics and Applied Biotechnology, Bangalore , India. After his Masters, he earned his PhD from Wageningen University, The Netherlands, on applied machine learning and data analysis. After his PhD he moved to Lyon, France, for his post-doctoral study with Synergie Lyon Cancer Centre as a Biostatistician where he extensively worked on big data analytics, cloud computing. After his post-doctoral study he was offered a scientist position with BASF Cropdesign, Belgium.

Before joining University of Birmingham and Queen Elizabeth Hospital he was working with University of Cambridge, Cambridge, UK focusing on metabolic driven diseases like Obesity, T2-diabetis using high throughput metabolomics, lipidomics technologies. His research interests includes integrative data analytics, predictive biomarker discovery, bioinformatics methods for diagnostics and network biology. Throughout his career, he was offered many fellowships from British Council, Dutch Government and Newton fellowships. He has published many papers in the international journals and actively collaborate with many Universities, for example Harvard University and University of Cambridge. So far, he has published 83 papers, and his h index is 25 based on google scholar.

# Acknowledgement<br>
This material is based upon work partially supported by the National Science Foundation under Grant No. 2302637. Any opinions, findings, and conclusions or recommendations expressed in this material are those of the author(s) and do not necessarily reflect the views of the National Science Foundation. Additionally, we would like to acknowldge the ISCB-ASBCB; and Meharry Medical College, School of Applied Computational Sciences.
